proxy/controller-grpc/src/lib.rs

```diff
@@ -193,9 +191,8 @@ impl<'a> From<&'a ::std::net::SocketAddr> for common::TcpAddress {
 
 // ===== impl common::scheme::Type =====
 
-impl<'a> TryInto<String> for &'a common::scheme::Type {
-    type Err = InvalidScheme;
-    fn try_into(self) -> Result<String, Self::Err> {
+impl common::scheme::Type {
+    pub fn try_as_string(&self) -> Result<String, InvalidScheme> {
```

Comment:
This should be `try_to_string`. `as_` is for things that are cheap (free), `to_` is for things that are expensive (e.g. allocating).